src/IO/Primitive.agda

```diff
@@ -1,38 +1,15 @@
 ------------------------------------------------------------------------
 -- The Agda standard library
 --
--- Primitive IO: simple bindings to Haskell types and functions
+-- This module is DEPRECATED. Please use IO.Primitive.Core instead
 ------------------------------------------------------------------------
 
--- NOTE: the contents of this module should be accessed via `IO`.
-
 {-# OPTIONS --cubical-compatible #-}
 
 module IO.Primitive where
 
-open import Level using (Level)
-private
-  variable
-    a : Level
-    A B : Set a
-
-------------------------------------------------------------------------
--- The IO monad
-
-open import Agda.Builtin.IO public
-  using (IO)
-
-infixl 1 _>>=_
-
-postulate
-  pure : A → IO A
-  _>>=_  : IO A → (A → IO B) → IO B
-
-{-# COMPILE GHC pure = \_ _ -> return    #-}
-{-# COMPILE GHC _>>=_  = \_ _ _ _ -> (>>=) #-}
-{-# COMPILE UHC pure = \_ _ x -> UHC.Agda.Builtins.primReturn x #-}
-{-# COMPILE UHC _>>=_  = \_ _ _ _ x y -> UHC.Agda.Builtins.primBind x y #-}
+open import IO.Primitive.Core public
 
--- Haskell-style alternative syntax
-return : A → IO A
-return = pure
+{-# WARNING_ON_IMPORT
+"IO.Primitive was deprecated in v2.1. Use IO.Primitive.Core instead."
+#-}
```

Comment:
@gallais: Are there plans to use `IO.Primitive` for something else?
(Otherwise, why the deprecation?)
(Maybe the end user want to import the "top-level" modules rather than `Core` modules.)

Reply:
No plans on my end. It was a cleanup because having `IO.Primitive` sit "above" other primitive-binding
modules like `IO.Primitive.Finite` seemed at odds with our current practice of the `X.Y` module being
a partial re-export of `X.Y.Z` ones.

It's named `Core` because we don't expect anyone to be importing it (or any of the `IO.Primitive` modules)
for that matter as we provide a high-level `IO` interface.